src/eda_server/project.py

```diff
@@ -375,3 +316,238 @@ async def tar_project(db: AsyncSession, large_data_id: int, project_dir: str):
             with open(tarfile_name, "rb") as f:
                 while data := f.read(CHUNK_SIZE):
                     await lobject.write(data)
+
+
+async def import_project_file(db: AsyncSession, file, file_type, project_id):
```

Comment:
Please add some type annotations. It's completely unclear what the `file` is and it's even more unexpected that this is a tuple of `(directory, filename)`.